### PR TITLE
Clarify guide docs around userId and jwt

### PR DESCRIPTION
**Pull request by alavers**
- Cleaned up the "Authenticating users" section and code samples
- Got rid of multi user clients guide and merged its content with
"Authenticating users" section
- Added a brief section in Web, iOS and Android guides linking to the
"Authenticating users" guide.
- Updated & clarified references in REST docs

### CODE REVIEW

File: pages/guide/native-android-sdk.md [hunk 1]
```diff
@@ -140,6 +140,13 @@ settings.setFileProviderAuthorities(authoritiesString);
 Smooch.init(this, settings, myInitCallback);
 ```
 
+## Login with userId and JWT
+
+After Smooch has initialized your user is ready to chat, and they will show up on the business side as a new Smooch appUser. However, these appUsers will not yet be associated to any user record you might have in an existing user directory.
```
Comment (mspensieri) [inline pages/guide/native-android-sdk.md]: > After Smooch has initialized your user is ready to chat, and they will show up on the business side as a new Smooch appUser.

It's not really after init, it's after the user sends their first message that they will be created

Comment (alavers) [reply pages/guide/native-android-sdk.md]: Good call :3 I'll clarify

File: pages/guide/authenticating-users.md [hunk 2]
```diff
@@ -40,29 +44,29 @@ To issue JWTs:
 
 1. Specify the JWT when calling `login` on the client:
 
-    Objective-C:
+    Web (Javascript):
+    ```javascript
+    Smooch.login("user-id", "jwt")
```
Comment (lemieux) [inline pages/guide/authenticating-users.md]: Do we mention anywhere that calling `login` must happen after `init` ?

Comment (alavers) [reply pages/guide/authenticating-users.md]: We do in the linked-from docs, but it's definitely a good idea to re-clarify that here 👍

**Comment by alavers** [comment]
@jpjoyal

File: pages/guide/authenticating-users.md [hunk 3]
```diff
@@ -75,3 +81,58 @@ To issue JWTs:
...
+
+## Users on multiple clients
+
+You may have a single user logging in as the same `userId` from multiple clients. For example, they have your app installed on both their iPhone and their iPad. You might also have Smooch integrated in both your mobile app as well as on your web site.
+
+Once a user has been logged in to Smooch, they will see the same conversation across each of these clients.
+
+## Omitting the userId
+
+Smooch will work perfectly fine without a `userId`. Profile information can still be included, and the user can take advantage of all rich messaging features, but the user will only be able to access the conversation from the client they're currently using. This means that even if you specify the same profile information, if the same individual opens Smooch on a new client, or runs your web app in an incognito browser session, they will see a newly created empty conversation when they open Smooch. Only the `userId` can be used to synchronize a conversation across clients.
```
Comment (jpjoyal) [inline pages/guide/authenticating-users.md]: > they will see a newly created empty conversation when they open Smooch
... and will be represented as two distinct `appUsers`

File: pages/guide/authenticating-users.md [hunk 4]
```diff
@@ -75,3 +81,58 @@ To issue JWTs:
...
+
+A `userId` can also be omitted at first and added at a later time. If you deploy an update to your app that assigns an existing user with a new `userId` that they didn't have before, any existing conversation history they have will be preserved and their messages will start being synchronized across all clients where that `userId` is being used. This is particularly useful if a user opens Smooch and starts a conversation before having logged in to your app or website.
```
Comment (jpjoyal) [inline pages/guide/authenticating-users.md]: > starts a conversation before having logged in to your app or website.
Worth adding OTTs in this sentence ?

File: pages/guide/authenticating-users.md [hunk 5]
```diff
@@ -75,3 +81,58 @@ To issue JWTs:
...
+
+## Switching users
+
+If your app allows a shared client to switch between multiple user identities you can call the `login` API multiple times to switch between different `userId`s.
+
+## Logging out
+
+Your app may have a logout function which brings users back to a login screen. In this case you would want to revert Smooch to a pre-login state. You can do this by calling the `logout` API.
+
+Calling `logout` will disconnect your user from any `userId` they were previously logged in with and it will remove any conversation history stored on the client. Logging out will *not* disable Smooch. While logged out, the user is free to start a new conversation but they will show up as if `userId` was omitted during Smooch initialization.
```
Comment (jpjoyal) [inline pages/guide/authenticating-users.md]: > but they will show up as if `userId` was omitted
I think we should say what happens in this case. Maybe just they will be represented as a separated appuser

File: pages/guide/native-ios-sdk.md [hunk 6]
```diff
@@ -147,7 +147,13 @@ Find a suitable place in your app's interface to invoke Smooch and use the code
 Smooch.show()
 ```
 
-### Updating Smooch
+## Login with userId and JWT
+
+After Smooch has initialized your user can start sending messages right away. These messages will show up on the business side under a new Smooch appUser. However, these appUsers will not yet be associated to any user record you might have in an existing user directory.
+
+If your application has a login flow, or if a user needs to access the same conversation from multiple devices, this is where `Smooch.login` comes into play. You can associate Smooch users with your own user directory by assigning them a `userId`. You will then issue each user a `jwt` credential during the login flow. You can read more about this in the [Authenticating users](/guide/authenticating-users/) section.
```
Comment (jpjoyal) [inline pages/guide/native-ios-sdk.md]: nit `Smooch.login` works only for swift. Maybe rephrase to "this is where the [login](https://docs.smooch.io/api/ios/Classes/Smooch.html#//api/name/login:jwt:completionHandler:) method comes into play"

File: pages/guide/authenticating-users.md (hunk 4 above)
Comment (alavers) [reply pages/guide/authenticating-users.md]: Sounds like a more advanced channel linking use case that better fits under channel linking/transfer guides.
